homeassistant/components/media_player/sonos.py

```diff
@@ -608,6 +613,13 @@ def update(self):
             else:
                 playlist_size = int(playlist_size)
 
+            if playlist_size is None or playlist_size == 0:
```

Comment:
Understandable, as I felt the same way at first.

The Sonos controller just says "No music", and the play/pause button does nothing on the surface. From a UPnP standpoint, it's valid for the SoCo package to throw the 701 error as playing or pausing, etc. on an empty playlist is an invalid state transition (because there is nothing to play or pause). So the options were to try/catch the 701 error everywhere, or control the play/pause support based on playlist size.